Classes/Middleware/BackendUserRedirectToFrontend.php

```diff
@@ -0,0 +1,69 @@
+<?php
+declare(strict_types=1);
+
+namespace TYPO3\CMS\FrontendEditing\Middleware;
+
+use Psr\Http\Message\ResponseInterface;
+use Psr\Http\Message\ServerRequestInterface;
+use Psr\Http\Message\UriInterface;
+use Psr\Http\Server\MiddlewareInterface;
+use Psr\Http\Server\RequestHandlerInterface;
+use TYPO3\CMS\Core\Authentication\BackendUserAuthentication;
+use TYPO3\CMS\Core\Http\RedirectResponse;
+
+/**
+ * Middleware check if backend user allowed to access BE
+ *
+ * @package TYPO3\CMS\FrontendEditing\Middleware
+ */
+class BackendUserRedirectToFrontend implements MiddlewareInterface
+{
+    /**
+     * Check if BE user is forced to see only FE
+     *
+     * @inheritDoc
+     */
+    public function process(ServerRequestInterface $request, RequestHandlerInterface $handler): ResponseInterface
+    {
+        $user = $this->getBackendUser();
+        if ($this->isUserDisallowedAccessBackend($user)) {
```

Comment:
Disable for admins. We don't want to lock admins out of the backend.

```suggestion
        if ($this->isUserDisallowedAccessBackend($user) && !$GLOBALS['BE_USER']->isAdmin()) {
```

Reply:
check for admin was added